### PR TITLE
Added wrapper for assembly-stats

### PR DESCRIPTION
**Pull request by maxlcummins**
### Description

Added a wrapper for assembly-stats - a tool for generating summary statistics on genomic assemblies

### QC

For all wrappers added by this PR, I made sure that

* [x] there is a test case which covers any introduced changes,
* [x] `input:` and `output:` file paths in the resulting rule can be changed arbitrarily,
* [x] all `environment.yaml` specifications follow [the respective best practices](https://stackoverflow.com/a/64594513/2352071),
* [x] wherever possible, command line arguments are inferred and set automatically (e.g. based on file extensions in `input:` or `output:`),
* [x] all fields of the example rules in the `Snakefile`s and their entries are explained via comments (`input:`/`output:`/`params:` etc.),
* [x] `stderr` and/or `stdout` are logged correctly (`log:`), depending on the wrapped tool,
* [x] the `meta.yaml` contains a link to the documentation of the respective tool or command,
* [x] `Snakefile`s pass the linting (`snakemake --lint`),
* [x] `Snakefile`s are formatted with [snakefmt](https://github.com/snakemake/snakefmt),
* [x] Python wrapper scripts are formatted with [black](https://black.readthedocs.io).

### CODE REVIEW

**Comment by maxlcummins** [comment]
An error popped up with the meta.yaml formatting - this should now be fixed.

File: bio/assembly-stats/test/Snakefile [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+        # If you want to add multiple options just delimit them with a space.
+        # Note that you can only pick one output format
+        # Check https://github.com/sanger-pathogens/assembly-stats for more details
+        extra="-t",
+    log:
+        "logs/{sample}.assembly-stats.log",
+    threads: 1
+    wrapper:
+        #"0.2.0/bio/assembly-stats"
+        "https://raw.githubusercontent.com/maxlcummins/snakemake-wrappers/assembly-stats/bio/assembly-stats/wrapper.py"
```
Comment (jafors) [inline bio/assembly-stats/test/Snakefile:28]:
```suggestion
        "master/bio/assembly-stats"
```

File: bio/assembly-stats/environment.yaml [hunk 2]
```diff
@@ -0,0 +1,8 @@
+channels:
+  - bioconda
+  - conda-forge
+dependencies:
+  - _libgcc_mutex=0.1=main
+  - assembly-stats=1.0.1=hc9558a2_3
+  - libgcc-ng=9.1.0=hdf63c60_0
+  - libstdcxx-ng=9.1.0=hdf63c60_0
```
Comment (jafors) [inline bio/assembly-stats/environment.yaml:8]: We only need to give the `assembly-stats` dependency here. The libraries should be installed with that automatically.
Also, you don't need to fix the version this strictly. ` ==1.0.1` is fine. Actually, fixing the major and minor version number (so: 1.0) is mostly sufficient.

File: bio/assembly-stats/wrapper.py [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+from os import path
+
+log = snakemake.log_fmt_shell(stdout=False, stderr=True)
+
+shell(
+    "assembly-stats"
+    " {snakemake.params.extra}"
+    " {snakemake.input.assembly}"
+    " >"
+    " {snakemake.output.assembly_stats}"
```
Comment (jafors) [inline bio/assembly-stats/wrapper.py:16]:
```suggestion
    " > {snakemake.output.assembly_stats}"
    " {log}"
```

You don't need to insert line breaks for every space. ;) Also, adding the `log`.

**Review by jafors — CHANGES_REQUESTED** [review]
Welcome to the contributors list! Good first try, thanks for that!
I have a few remarks below.

File: bio/assembly-stats/environment.yaml (hunk 2 above)
Comment (maxlcummins) [reply bio/assembly-stats/environment.yaml]: Hi Jan! Thanks a lot for your warm welcome and your guidance. I have streamlined the environment.yaml and implemented the other changes you suggested. Hopefully now it's ready to be merged :)

**Review by jafors — APPROVED** [review]
Great, that's good to go!

**Pull request merged**